### ISSUE TITLE
The 'maximize = True' option in  xgboost.cv don't work

### ISSUE THREAD

**Issue by lyhue1991** [dmlc/xgboost#3811]
I  want to  use an user-defined metric by  'feval'  parameter in xgboost.cv, but it seems that the 'maximize = True ' option don't work at all.  It seems that it's a bug.

```{python}
from __future__ import print_function
import numpy as np
import pandas as pd
import xgboost
from sklearn import datasets
from sklearn.model_selection import train_test_split

data,label = datasets.make_classification(n_samples= 10000, n_features=20, n_informative= 6 ,
             n_classes=2, n_clusters_per_class=10,random_state=0)
X_train,X_test,y_train,y_test = train_test_split(data,label)

# custom metric: ks, it is a score, larger is better
from scipy import stats
def ks(preds,xgbtrain):
    label = xgbtrain.get_label()
    df = pd.DataFrame(data = np.array([preds,label]).T,columns = ['preds','label'])
    df_0,df_1 = df[df['label']<0.5],df[df['label']>=0.5]
    ks,ks_pvalue = stats.ks_2samp(df_0['preds'].values,df_1['preds'].values)
    return 'ks',ks

dtrain = xgboost.DMatrix(X_train,label = y_train)
dtest = xgboost.DMatrix(X_test,label = y_test)

param = {'max_depth': 2, 'eta': 0.1, 'silent':1, 'objective':'binary:logistic'}

# use xgboost.cv to use the ks as the custom evaluation but  'maximize = True' don't work
cvresult = xgboost.cv(param, dtrain, num_boost_round = 1000,
                  nfold=5, seed=10, metrics = 'auc' ,feval = ks, maximize = True,
                  callbacks=[ xgboost.callback.print_evaluation(show_stdv=False),
                             xgboost.callback.early_stop(10) ])

num_round_best = cvresult.shape[0] - 1 
```
the results as below:

```
[0]	train-auc:0.674289	train-ks:0.238805	test-auc:0.665216	test-ks:0.234206
Multiple eval metrics have been passed: 'test-ks' will be used for early stopping.

Will train until test-ks hasn't improved in 10 rounds.
[1]	train-auc:0.680778	train-ks:0.248258	test-auc:0.669496	test-ks:0.235578
[2]	train-auc:0.692075	train-ks:0.273699	test-auc:0.679234	test-ks:0.260029
[3]	train-auc:0.697084	train-ks:0.281107	test-auc:0.686079	test-ks:0.267151
[4]	train-auc:0.700154	train-ks:0.281611	test-auc:0.687756	test-ks:0.267845
[5]	train-auc:0.707491	train-ks:0.291344	test-auc:0.69528	test-ks:0.284074
[6]	train-auc:0.710668	train-ks:0.293191	test-auc:0.696747	test-ks:0.285947
[7]	train-auc:0.716494	train-ks:0.309238	test-auc:0.702744	test-ks:0.294814
[8]	train-auc:0.719062	train-ks:0.318497	test-auc:0.704835	test-ks:0.299245
[9]	train-auc:0.72203	train-ks:0.324148	test-auc:0.707431	test-ks:0.310519
[10]	train-auc:0.72729	train-ks:0.33985	test-auc:0.712009	test-ks:0.320517
Stopping. Best iteration:
[0]	train-auc:0.674289+0.00230273	train-ks:0.238805+0.00439313	test-auc:0.665216+0.00744289	test-ks:0.234206+0.0127197

```

**Comment by hcho3** [comment]
You should pass `maximize` to `xgboost.callback.early_stop`, since you are manually adding it to `callbacks`:
```python
cvresult = xgboost.cv(param, dtrain, num_boost_round = 1000,
                  nfold=5, seed=10, metrics = 'auc' ,feval = ks, maximize = True,
                  callbacks=[xgboost.callback.print_evaluation(show_stdv=False),
                             xgboost.callback.early_stop(10, maximize=True) ])
```

See the documentation at https://xgboost.readthedocs.io/en/latest/python/python_api.html#xgboost.callback.early_stop

**Comment by lyhue1991** [comment]
Thanks, I got it.
